### PR TITLE
r/aws_connect_queue

### PR DESCRIPTION
**Pull request by GlennChia**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #21021

Output from acceptance testing:

```
$ make testacc TESTS=TestAccConnectQueue_serial PKG=connect
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/connect/... -v -count 1 -parallel 20 -run='TestAccConnectQueue_serial'  -timeout 180m
=== RUN   TestAccConnectQueue_serial
=== RUN   TestAccConnectQueue_serial/update_max_contacts
    queue_test.go:190: A bug in the service API has been reported
=== RUN   TestAccConnectQueue_serial/update_outbound_caller_config
=== RUN   TestAccConnectQueue_serial/update_status
=== RUN   TestAccConnectQueue_serial/basic
=== RUN   TestAccConnectQueue_serial/disappears
    queue_test.go:94: Queues do not support deletion today
=== RUN   TestAccConnectQueue_serial/update_hours_of_operation_id
--- PASS: TestAccConnectQueue_serial (678.45s)
    --- SKIP: TestAccConnectQueue_serial/update_max_contacts (0.00s)
    --- PASS: TestAccConnectQueue_serial/update_outbound_caller_config (158.45s)
    --- PASS: TestAccConnectQueue_serial/update_status (162.19s)
    --- PASS: TestAccConnectQueue_serial/basic (155.96s)
    --- SKIP: TestAccConnectQueue_serial/disappears (0.00s)
    --- PASS: TestAccConnectQueue_serial/update_hours_of_operation_id (201.85s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/connect    686.313s

...
```

### CODE REVIEW

File: website/docs/r/connect_queue.html.markdown [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+---
+
+# Resource: aws_connect_queue
+
+Provides an Amazon Connect Queue resource. For more information see
+[Amazon Connect: Getting Started](https://docs.aws.amazon.com/connect/latest/adminguide/amazon-connect-get-started.html)
+
+~> **NOTE:** Due to The behaviour of Amazon Connect you cannot delete queues.
+
+## Example Usage
```
Comment (AdamTylerLynch) [inline website/docs/r/connect_queue.html.markdown:16]: I would like to see two more examples that include Caller Confit and Quick Connect Ids for completeness.

File: website/docs/r/connect_queue.html.markdown [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+description: |-
+  Provides details about a specific Amazon Connect Queue
+---
+
+# Resource: aws_connect_queue
+
+Provides an Amazon Connect Queue resource. For more information see
+[Amazon Connect: Getting Started](https://docs.aws.amazon.com/connect/latest/adminguide/amazon-connect-get-started.html)
+
+~> **NOTE:** Due to The behaviour of Amazon Connect you cannot delete queues.
```
Comment (AdamTylerLynch) [inline website/docs/r/connect_queue.html.markdown:14]: Nice job including this note for the users. Having thorough documentation is important.

**Review by AdamTylerLynch — CHANGES_REQUESTED** [review]
One minor change requested in the docs, asking for a few more examples.

File: website/docs/r/connect_queue.html.markdown (hunk 1 above)
Comment (GlennChia) [reply website/docs/r/connect_queue.html.markdown:16]: Example with `quick_connect_ids` resolved in 5942b3be5871a3342640612c1d920efedd1d2a85
Example with `outbound_caller_config` resolved in ebf5333e70f5b1ab524d0d05841c48d61f4003fd

**Review by AdamTylerLynch — APPROVED** [review]
LGTM 🚀

**Comment by breathingdust** [comment]
LGTM 🚀  Thanks @GlennChia!

Validated Acceptance Tests in Commercial (us-west-2)

```console
make testacc TESTS=TestAccConnectQueue_serial PKG=connect
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/connect/... -v -count 1 -parallel 20 -run='TestAccConnectQueue_serial'  -timeout 180m
=== RUN   TestAccConnectQueue_serial
=== RUN   TestAccConnectQueue_serial/basic
=== RUN   TestAccConnectQueue_serial/disappears
    queue_test.go:89: Queues do not support deletion today
=== RUN   TestAccConnectQueue_serial/update_hours_of_operation_id
=== RUN   TestAccConnectQueue_serial/update_max_contacts
    queue_test.go:182: A bug in the service API has been reported
=== RUN   TestAccConnectQueue_serial/update_outbound_caller_config
=== RUN   TestAccConnectQueue_serial/update_status
--- PASS: TestAccConnectQueue_serial (427.78s)
    --- PASS: TestAccConnectQueue_serial/basic (104.24s)
    --- SKIP: TestAccConnectQueue_serial/disappears (0.00s)
    --- PASS: TestAccConnectQueue_serial/update_hours_of_operation_id (119.52s)
    --- SKIP: TestAccConnectQueue_serial/update_max_contacts (0.00s)
    --- PASS: TestAccConnectQueue_serial/update_outbound_caller_config (100.25s)
    --- PASS: TestAccConnectQueue_serial/update_status (103.77s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/connect	432.281s
```

Validated Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TESTS=TestAccConnectQueue_serial PKG=connect
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/connect/... -v -count 1 -parallel 20 -run='TestAccConnectQueue_serial'  -timeout 180m
=== RUN   TestAccConnectQueue_serial
=== RUN   TestAccConnectQueue_serial/update_max_contacts
    queue_test.go:182: A bug in the service API has been reported
=== RUN   TestAccConnectQueue_serial/update_outbound_caller_config
=== RUN   TestAccConnectQueue_serial/update_status
=== RUN   TestAccConnectQueue_serial/basic
=== RUN   TestAccConnectQueue_serial/disappears
    queue_test.go:89: Queues do not support deletion today
=== RUN   TestAccConnectQueue_serial/update_hours_of_operation_id
--- PASS: TestAccConnectQueue_serial (373.67s)
    --- SKIP: TestAccConnectQueue_serial/update_max_contacts (0.00s)
    --- PASS: TestAccConnectQueue_serial/update_outbound_caller_config (105.15s)
    --- PASS: TestAccConnectQueue_serial/update_status (84.96s)
    --- PASS: TestAccConnectQueue_serial/basic (86.20s)
    --- SKIP: TestAccConnectQueue_serial/disappears (0.00s)
    --- PASS: TestAccConnectQueue_serial/update_hours_of_operation_id (97.35s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/connect	376.804s
```

**Pull request merged**